Kernel/Memory/SharedInodeVMObject.cpp

```diff
@@ -51,12 +51,17 @@ SharedInodeVMObject::SharedInodeVMObject(SharedInodeVMObject const& other, Fixed
 
 ErrorOr<void> SharedInodeVMObject::sync(off_t offset_in_pages, size_t pages)
 {
-    SpinlockLocker locker(m_lock);
-
     size_t highest_page_to_flush = min(page_count(), offset_in_pages + pages);
 
     for (size_t page_index = offset_in_pages; page_index < highest_page_to_flush; ++page_index) {
-        auto& physical_page = m_physical_pages[page_index];
+        // FIXME: This will be locked multiple times which is probably bad for
```

Comment:
Isn't this also racy now? How do we stop two threads from grabbing the same physical page?

Reply:
I added a Mutex to solve this now. Still, we lock both the inode Mutex and SharedInode Mutex, which could be OK, but if the syscall of `msync` is called, it will take there a Spinlock of the `address_space()` and then it will call the `sync()` method on the shared inode, which is quite bad thing. I will try to fix this in a future patch though.

Reply:
I fixed this now in the last (fourth) commit in this PR. We grab a Mutex now to ensure that we can never have two threads grabbing the same physical page. The next future PR will remove the big lock from all of the memory-related syscalls.